src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,32 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core throws a PlatformNotSupportedException for deserializing many exceptions.
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
+            {
+                return;
+            }
+
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
+            info.AddValue("ClassName", "ClassName");
+            info.AddValue("Message", "Message");
+            info.AddValue("InnerException", null);
+            info.AddValue("HelpURL", null);
+            info.AddValue("StackTraceString", null);
+            info.AddValue("RemoteStackTraceString", null);
+            info.AddValue("RemoteStackIndex", 5);
+            info.AddValue("HResult", 5);
+            info.AddValue("Source", null);
+            info.AddValue("ExceptionMethod", null);
+
+            ConstructorInfo constructor = typeof(T).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[] { typeof(SerializationInfo), typeof(StreamingContext) }, null);
```

Comment:
@morganbr 

I assume the reducer still hasn't been updated to recognize this api pattern...

@hughbe - I think you'd want to scope this to exceptions whose serialization constructor is public or protected and exposed in the ref assembly - otherwise, I'd expect we'd just remove the constructor eventually. (And the .NET Native platform won't let you reflect on them if they're not in the ref assembly.)